tests/30rooms/70publicroomslist.pl

```diff
@@ -0,0 +1,85 @@
+test "Name/topic keys are corrct",
+   requires => [ $main::API_CLIENTS[0], local_user_fixture() ],
+
+   check => sub {
+      my ( $http, $user ) = @_;
+
+      my %rooms = (
+         publicroomalias_no_name => {},
+         publicroomalias_with_name => {
+            name => "name_1",
+         },
+         publicroomalias_with_topic => {
+            topic => "topic_1",
+         },
+         publicroomalias_with_name_topic => {
+            name => "name_2",
+            topic => "topic_2",
+         },
+      );
+
+      Future->needs_all( map {
+         my $alias_local = $_;
+         my $room = %rooms->{$alias_local};
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => $alias_local,
+            name => $room->{name},
+            topic => $room->{topic},
+         )
+      } keys %rooms )
+      ->then( sub {
+         $http->do_request_json(
+            method => "GET",
+            uri    => "/r0/publicRooms",
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         log_if_fail "publicRooms", $body;
+
+         assert_json_keys( $body, qw( start end chunk ));
+         assert_json_list( $body->{chunk} );
+
+         my %seen = map {
+            $_ => 0,
+         } keys ( %rooms );
+
+         foreach my $room ( @{ $body->{chunk} } ) {
+            assert_json_keys( $room,
+               qw( world_readable guest_can_join num_joined_members aliases )
+            );
+
+            my $name = $room->{name};
+            my $topic = $room->{topic};
+            my $canonical_alias = $room->{canonical_alias};
+
+            my $aliases = $room->{aliases};
+            foreach my $alias ( @{$aliases} ) {
+               foreach my $alias_local ( keys %rooms ) {
+                  if ( $alias =~ m/^\Q#$alias_local:\E/ ) {
+                     my $room_config = %rooms->{$alias_local};
+
+                     log_if_fail "Alias", $alias_local;
+                     log_if_fail "Room", $room;
+
+                     assert_eq( $canonical_alias, $alias );
+                     assert_eq( $room->{num_joined_members}, 1 );
+
+                     $room_config->{name} eq $name or die "Unexpected name";
+                     $room_config->{topic} eq $topic or die "Unexpected name";
```

Comment:
Hmm.. Rare for that kind of pattern to come up, actually. Offhand the only suggestion I can really think of is just manually splitting the cases;

    if( defined $topic ) {
       assert_eq( $room_config->{topic}, $topic, 'room topic' );
    }
    else {
       defined $room_ocnfig->{topic} and die "Expected not to find a topic";
    }

If that kind of pattern comes up a lot we might consider an `assert_eq_or_undef` or somesuch